notebooks/6_Putting_it_All_Together.ipynb

```diff
@@ -0,0 +1,1002 @@
+    "# data = np.load(kpcovr_file)\n",
+    "# T_train = data['T_train']\n",
+    "# T_test = data['T_test']"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Compare the predictive power of KPCovR and KPCA components"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 34,
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "The regression on 5 KPCovR principal components has an RMSE of 14.607ppm on the testing set.\n",
+      "By comparison, a regression on 5 KPCA components gives an RMSE of 64.1938ppm.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# r = RidgeCV(cv=4, fit_intercept=False, alphas=np.logspace(-8, 2, 10))\n",
+    "# r.fit(T_train, Y_train)\n",
+    "\n",
+    "# Yp_test = r.predict(T_test)\n",
+    "# rmse = calc_rmse(Yp_test, Y_test, y_scaler)\n",
+    "\n",
+    "# print(\n",
+    "#     f\"The regression on {kpcovr.n_components} KPCovR principal components has an RMSE of {rmse}ppm on the testing set.\"\n",
+    "# )\n",
+    "\n",
+    "# from sklearn.decomposition import KernelPCA\n",
+    "\n",
+    "# kpca = KernelPCA(n_components=kpcovr.n_components, kernel=\"linear\")\n",
+    "# kpca.fit(Phi_train)\n",
+    "# T_train_kpca = kpca.transform(Phi_train)\n",
+    "# T_test_kpca = kpca.transform(Phi_test)\n",
+    "\n",
+    "# r.fit(T_train_kpca, Y_train)\n",
+    "# Yp_kpca_test = r.predict(T_test_kpca)\n",
+    "# rmse = calc_rmse(Yp_kpca_test, Y_test, y_scaler)\n",
+    "\n",
+    "# print(\n",
+    "#     f\"By comparison, a regression on {kpcovr.n_components} KPCA components gives an RMSE of {rmse}ppm.\"\n",
+    "# )"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Exporting to a Chemiscope"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 29,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import chemiscope\n",
+    "\n",
+    "output_name = \"kpcovr-chemiscope.json\""
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "We'll write one output combining the training set and the testing set."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 32,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "T = np.zeros((X_raw.shape[0], kpcovr.n_components))\n",
+    "T[i_test] = T_test\n",
+    "T[i_train] = T_train\n",
+    "\n",
+    "train_test = np.zeros(X_raw.shape[0])\n",
+    "train_test[i_train] = 1\n",
+    "\n",
+    "is_active = np.zeros(X_raw.shape[0])\n",
+    "is_active[i_train[sample_idx]] = 1\n",
+    "\n",
+    "n_cur = 10\n",
+    "cur_features = np.zeros((X_raw.shape[0], n_cur))\n",
+    "cur_features[i_test] = X_raw_test[:, cur_idx[:n_cur]]\n",
+    "cur_features[i_train] = X_raw_train[:, cur_idx[:n_cur]]\n",
+    "\n",
+    "chemiscope_properties = {\n",
+    "    **{\n",
+    "        f\"kpcovr_{i}\": {\"target\": target, \"values\": T[:, i]}\n",
```

Comment:
this is done automatically in chemiscope.write_input if you pass

```py
chemiscope_properties = {
    "kpcovr": {
        "target": "atom", 
        "values": T[:, :kpcovr.n_components]
    },
    "cur_feature": {
        "target": "atom", 
        "values": cur_features[:, :n_cur]
   },
    **{
        prop_name: {"target": target, "values": Y[:, i]}
        for i, prop_name in enumerate(properties)
    },
}
```

Reply:
Addressed in most recent commit!